docs/source/reference/upgrade/from-router-v1.mdx

```diff
@@ -4,230 +4,270 @@ subtitle: Upgrade from version 1.x to 2.x of GraphOS Router
-      # If one of these headers matches we will log supergraph and subgraphs requests/responses
-      experimental_when_header: # REMOVED
-        - name: apollo-router-log-request
-          value: my_client
-          headers: true # default: false
-          body: true # default: false
-```
+**Upgrade step**: replace uses of `--schema` with `router config schema` to print the configuration supergraph.
 
-This feature no longer exists and can be replaced with another one included in custom telemetry. Here is how you would configure custom telemetry to achieve the same result:
 
-```yaml title="router.yaml"
-telemetry:
-  instrumentation:
-    events:
-      router:
-        request: # Display router request log
-          level: info
-          condition:
-            eq:
-              - request_header: apollo-router-log-request
-              - my_client
-        response: # Display router response log
-          level: info
-          condition:
-            eq:
-              - request_header: apollo-router-log-request
-              - my_client
-      supergraph:
-        request: # Display supergraph request log
-          level: info
-          condition:
-            eq:
-              - request_header: apollo-router-log-request
-              - my_client
-        response:
-          level: info
-          condition:
-            eq:
-              - request_header: apollo-router-log-request
-              - my_client
-      subgraph:
-        request: # Display subgraph request log
-          level: info
-          condition:
-            eq:
-              - supergraph_request_header: apollo-router-log-request
-              - my_client
-        response: # Display subgraph response log
-          level: info
-          condition:
-            eq:
-              - supergraph_request_header: apollo-router-log-request
-              - my_client
-```
+### Removed automatically updating configuration at runtime
 
-### Introspection depth limit
+The ability to automatically upgrade configurations at runtime is removed. Previously, during configuration parsing/validation, the router 'upgrade migrations' would be applied automatically to generate a valid runtime representation of a config for the life of the executing process.
 
-The [schema-intropsection schema](https://spec.graphql.org/draft/#sec-Schema-Introspection.Schema-Introspection-Schema) is recursive: a client can query the fields of the types of some other fields, and so on arbitrarily deep. This can produce responses that grow much faster than the size of the request.
+Automatic configuration upgrades can still be applied explicitly.
 
-To protect against abusive requests Router now refuses to execute introspection queries that nest list fields too deep and returns an error instead. The criteria matches `MaxIntrospectionDepthRule` in graphql-js, but may change in future versions.
+**Upgrade step**: use the `router config` commands as shown at the top of the upgrade guide.
```

Comment:
Does this mean the `router config upgrade` command will perform all the migrations in place and customers will essentially have updated their config file permanently. Is there an additional step they need here?

Reply:
More or less. The exact steps are documented near the top of the upgrade guide. It applies all the automatic migrations to the file, and we expect users to follow the upgrade guide *after* doing this step. I wasn't sure how to work that in, it's a bit awkward that we're referencing this again when it's already at the top. I hope that can be addressed in a follow up : )